src/Microsoft.Health.Dicom.Functions/Indexing/Models/StartReindexActivityInput.cs

```diff
@@ -8,23 +8,19 @@
 namespace Microsoft.Health.Dicom.Functions.Indexing.Models
 {
     /// <summary>
-    /// Represents a batch of DICOM instances to be re-indexed based on the given tags.
+    ///  Represents input to <see cref="ReindexDurableFunction.StartReindexActivityAsync"/>
     /// </summary>
-    public class ReindexBatch
+    public class StartReindexActivityInput
     {
         /// <summary>
-        /// Gets or sets the inclusive starting watermark.
+        /// Gets or sets the operation id
         /// </summary>
-        public long Start { get; set; }
+        public string OperationId { get; set; }
 
         /// <summary>
-        /// Gets or sets the exclusive ending watermark.
+        /// Gets or sets the tag keys
         /// </summary>
-        public long End { get; set; }
+        public IEnumerable<int> TagKeys { get; set; }
```

Comment:
Generally, let's try to use some list-like interfaces. `IReadOnlyList<T>` when the users doesn't need to/should not modify it, `IList<T>` when we don't care about the implementation, and `List<T>` when it's used internally between our own private or internal methods/classes/etc

Reply:
Guess the reason is `List` kind types(`IList`, 'IReadonlyList','List') provides more capability, so makes code easier?
Also List is very commonly used when we don't know the size?